### ISSUE TITLE
implicit mensur

### ISSUE THREAD

**Issue by rettinghaus** [music-encoding/guidelines#29]
The section _5.2. Mensuration Signs_ is not totally clear on how the information given should be rendered, and I'd like to improve it. 
Q: Do `@modusmaior`, `@modusminor`, `@tempus` and `@prolatio` imply a sign to be rendered, or are they meant to *just* carry musical meaning, i.e. do one always have to give *explicit* information about the desired sign to be rendered by `@sign`, `@orient`, `@slash`, and `@dot`?

**Comment by lpugin** [comment]
Could you check this with the mensura-ig?

**Comment by bwbohl** [comment]
@rettinghaus could you please check if the current version better suits your ideas? This ticket could probably be closed

**Comment by rettinghaus** [comment]
@bwbohl I'm happy that `@sign` and `@dots` will be moved to the visual domain eventually. But still my question: does this mean, that a `<mensur>` without these attribute will still show up in a rendered score (based on the other attributes), or would this be an un-notated change in mensuration? @martha-thomae @giulianodibacco

**Comment by bwbohl** [comment]
@martha-thomae @giulianodibacco could any of you give us a heads up on this?

**Comment by martha-thomae** [comment]
Hi @rettinghaus and @bwbohl,

So I am still waiting for the answer on the mailing list to make the pull request where I have changed these attributes to the visual domain. Not many answers so far, but @giulianodibacco wanted to have some input in the mailing list before this proposal is evaluated by the technical committee. I think I will put some kind of deadline to answer this; otherwise, we are not moving forward. Anyway...

> does this mean, that a <mensur> without these attribute will still show up in a rendered score (based on the other attributes)

Currently, `<mensur>` without these attributes (`@sign` and `@dots`) is shown in the Verovio rendered score. I would assume that the idea of the distinction between the visual-domain attributes (`@sign` and `@dots`) and the logical-domain ones (i.e., the ones encoding the actual mensuration) is to allow for encoding the mensuration of a voice (or a section) where no mensuration sign is shown. Therefore, when the visual-domain attributes are missing, the rendered score would not show any mensuration sign.

> would this be an un-notated change in mensuration?
I have never seen changes in mensuration that are not indicated by a mensuration sign. (Does this actually happen?)

To be honest, now that I am working on the examples of the guidelines, I am a little conflicted about removing the mensuration-related attributes from `<staffDef>` (changes proposed in https://github.com/music-encoding/music-encoding/pull/541). These attributes allowed to encode the mensuration of a voice when no mensuration sign is given (which is a common situation in _Ars antiqua_, and in several pieces from the _Ars nova_). 

### Example:
- The code `<staffDef n="1" lines="5" notationtype="mensural" tempus="3" prolatio="3"/>` will encode a perfect tempus and major prolation without any mensuration sign being rendered. 
- The code `<staffDef n="1" lines="5" notationtype="mensural" tempus="3" prolatio="3" mensur.sign="O" mensur.dot="true"/>` will encode the same mensuration but with a mensuration sign showing as well. 

I include here an MEI file that shows this. All odd number staves only encode the mensuration in `<staffDef>`, and all even staves encode both the mensuration and the mensuration sign.

[proof_of_concept_mensuration.mei.zip](https://github.com/music-encoding/guidelines/files/4318716/proof_of_concept_mensuration.mei.zip)

The element `<mensur>` (as of now) seem to be used for encoding **mensuration changes** (within the stream of notes), which—as far as I know—have to be indicated by a mensuration sign. (Otherwise, how do you know that a change in mensuration occurred?)

**In conclusion,** because we are pushing into removing the mensuration-related attributes from the `<staffDef>` elements and leaving them only to be used in `<mensur>`, to encode a piece with no mensuration signs at all, **yes, the mensuration sign shouldn't be rendered if `@sign` or `@dots` are missing in `<mensur>`**. But I don't like this very much because that would also allow to encode a change in mensuration (when using `<mensur>` for the second, third, etc. time on a voice) with no mensuration sign being rendered, which I don't think is something that happens (Am I wrong?)

**Comment by giulianodibacco** [comment]
@bwbohl your observation is an excellent one. The original intention was simply to allow recording the presence of a specific sign with `@sign`, `@dots` etc. and keep it flexible. As a consequence, you are right, `<mensur>` without these attributes would mean that the mensuration is not explicitly noted in the score but is conjectural. The rendering of this makes me feel torn. On the one hand, we could decide that in absence of a specific `@sign` the score would _add_ the most common symbols for that particular `@tempus` / `@prolatio` pair. On the other hand, good editorial practice dictates that what is editorially added (by conjecture or calculation or contextual analysis or whatever) should be pointed out within square brackets. And I am not sure that I am able to decide what is the most common symbol for a certain `@divisio` nor to leave the processor the responsibility to add a stock symbol. 
I think that my inclination would be to say that no sign should be added by the processor unless it is explicitly encoded. But then what encoders should do, if they notice a change of mensuration, record it in `<mensur>` and want something indicating it in the score? How do I indicate that from here the mensuration as changed to as if there was a circle with a dot. I would add it within square brackets. This sounds as a `ges` somehow.

**Comment by giulianodibacco** [comment]
I agree with @martha-thomae that we should not wait forever that opinions are posted to the list. I just wanted to avoid that such a big change would be discussed in the GitHub repo only. (I often advocated the need to make clear the distinction between the decision-making process and the technical process that translate the decision into the ODD). I think that the lack of response or interest means that we can go ahead.

**Comment by martha-thomae** [comment]
> I agree with @martha-thomae that we should not wait forever that opinions are posted to the list. I just wanted to avoid that such a big change would be discussed in the GitHub repo only. (I often advocated the need to make clear the distinction between the decision-making process and the technical process that translate the decision into the ODD). I think that the lack of response or interest means that we can go ahead.

Perfect! Thanks for the confirmation, @giulianodibacco. I will make the PR moving those two attributes to the visual domain and with the additional `@divisio` and values of `@sign` for encoding Italian mensuration signs. Thanks again!


Do you have any thoughts on the other points of my comment? About `<mensur>` and `<staffDef>`? I would like to hear them :)

**Comment by giulianodibacco** [comment]
Good point @martha-thomae that we should consider all voice parts. From your examples I am even more convinced that most of the problem could be solved if we could encode the editorial addition of a sign. I am not convinced that going back to encoding the 'general mensuration' of the piece in `<staffDef>` solves the problem of having to decide that a certain mensuration is associated to a certain sign. For `@divisio` I wouldn't know how to decide which is the preferred sign to print on the score for 'senaria perfecta' since we have three. I must confess here that I don't have a clear perspective, in general, of how editorial changes are encoded/rendered.

**Comment by giulianodibacco** [comment]
I am eager to hear from Klaus @rettinghaus since I was always convinced that to notate `<mensur>` in the stream only solves a lot of problems and ambiguities. My understanding is that we are discovering that we need some adjustments to make it work; it looks to me that we just need to make sure that the score can reflect (when needed) the analytical process that made the encoder/editor decide what to encode in `<mensur>`.  If we solve this, for each voice and at any given point in the stream of notes the encoder will be able to represent EITHER the explicit presence of a sign and the mensural meaning of that sign, OR the mensuration that is implicit in the context and the sign that should be added editorially in the score to signify that. (Make sense?)

**Comment by bwbohl** [comment]
Thanks for your answers, as to the credits, should all go to @rettinghaus I'm just the issue pusher here ;-)

**Comment by giulianodibacco** [comment]
@bwbohl you are right, I mis-read the thread. Rephrase: @rettinghaus your observation is an excellent one... etc.

**Comment by rettinghaus** [comment]
@martha-thomae 

> Currently, <mensur> without these attributes (`@sign` and `@dots`) is shown in the Verovio rendered score.

It does, because I implemented it this way. :-) This is where the question originally came from. The main question is: do we want to allow the possibility to encode contradicting information in a single element, or should the encoder *always* have to use editorial elements like `<app>` ? As we allow it on other elements (and it is not controllable by schematron rules) I would guess that it's ok.

@giulianodibacco 
If a change in mensuration is conjectural and you want to make it visible in a rendered score, you probably need to use editorial elements. However, the need to indicate brackets or something similar it would best to make `mensur` a member of `att.enclosingChars`!

**Comment by giulianodibacco** [comment]
What about the following: `<mensur>` records mensuration in the stream of notes (at the beginning of a piece or whatever it changes) regardless of the presence of a sign (Verovio: no sign is displayed). When a sign is notated, it is explicitly recorded using the available attributes (Verovio: displays it accordingly). When the encoder is interested in displaying a conjectural sign, within square brackets or other symbols according to their editorial policy, then <app> should be used. 
If you @martha-thomae @rettinghaus agree on this, I will post a quick request of approval to the mensural list.

**Comment by rettinghaus** [comment]
@giulianodibacco Sounds fine to me.

**Comment by martha-thomae** [comment]
Sounds good to me too.

**Comment by lpugin** [comment]
Sorry, I must have overlooked something, but am I correct that /mensur was completely removed from staffDef? That it, both attributes and elements? I am doubtful about this. I think /mensur indications work pretty much like clefs. It can be specified in the staffDef and then be changed in the stream of data with /mensur (respectively /clef).

**Comment by lpugin** [comment]
I remember the discussion in Maryland. Maybe a compromise could be to drop mensur /staffDef and /scoreDef attributes, but still allow mensur as child. This could be useful for hybrid encodings that still use mensur but treats them similarly as meterSig. Here is one example
https://wiki.ccarh.org/wiki/MuseData_Example:_mensural_signs
Notice the sign repeated at the end of the staves. I would expect this type of rendering to be triggered with a /mensur within /staffDef and not within /layer.

**Comment by martha-thomae** [comment]
I also agree to go back to encoding the mensuration within the `<staffDef>` and `<scoreDef>` elements. I agree that this is more MEI-like and it is consistent with the encoding of meter in CMN. I do agree with the idea of encoding clefs and meter in `<staffDef>` because it is kind of the "voice's metadata," and that is also true for mensuration. Therefore, I would also like to revert these changes. 

Back when this change was discussed, I didn't have enough background on the technical aspects of MEI to have a strong argument in favor/opposition to this, and I think that the group in which this was discussed was very small (most people have already lef the meeting), were there enough people with technical expertise?

If the argument behind removing the mensuration-related attributes from `<staffDef>` (reduction of the "zoo" of attributes in `<staffDef>`) is not **that** important, then I am in favor on going back to having the mensuration encoded in the `<staffDef>`. I think that this is more intuitive—given how meter is encoded—and it is also more appropriate because we are defining a characteristic of the staff (the mensuration of the voice). Changes in mensuration can then be encoded with `<mensur>` elements within the layer.

If there are stronger arguments against this, then the middle-ground solution proposed by @lpugin is fine.

Thoughts?

**Comment by kepper** [comment]
I'm certainly no expert on this, but doesn't this relate to the handling of clefs (and other things) for CMN, as suggested by @lpugin in https://github.com/music-encoding/guidelines/issues/29#issuecomment-642075200? A clef can either be encoded as attributes within `<staffDef>` (which is more metadata-like), or as element inside `<layer>` (which is more event-like, or maybe even visual). It's perfectly possible to render a clef from `<staffDef>`, or to suppress its rendition by providing `@clef.visible="false"`. How does this differ from the mensuration questions over here? I'd be in favor of stepping back and revising both the original problem and the proposed solution, considering how they relate to other aspects of MEI. The proposed solution might be fine, but at least I don't understand why we're loosing consistency here (which is my perception of the current solution). I'm not in favor of dropping attributes just to get rid if a perceived "zoo" – some of these animals may not be appealing at first sight, but even the naked mole-rat knows some impressive tricks… To make a long story short, I'd follow @martha-thomae and revert…

**Comment by martha-thomae** [comment]
Thanks, @kepper. I will revert the changes (we can always discuss it later). 

I will make a PR for this (I don't like reverting in the main repo).

**Comment by bwbohl** [comment]
what's the correspondingg PR? #180?

**Comment by martha-thomae** [comment]
Hi @bwbohl. Sorry, I took about a week off.

The corresponding PR is https://github.com/music-encoding/music-encoding/pull/676. And I will make the appropriate changes in the guidelines (the development version, because these changes were in development) in a few days. Will let you know when it is there.

> what's the corresponding PR? #180?
This is correcting the current version of the guidelines (4.0.1) because it had some missing/misleading information.

**Comment by rettinghaus** [comment]
@martha-thomae @bwbohl May this be closed now?

**Comment by martha-thomae** [comment]
Hi @rettinghaus!

So, regarding the original issue:
> The section _5.2. Mensuration Signs_ is not totally clear on how the information given should be rendered, and I'd like to improve it.
> Q: Do `@modusmaior`, `@modusminor`, `@tempus` and `@prolatio` imply a sign to be rendered, or are they meant to _just_ carry musical meaning, i.e. do one always have to give _explicit_ information about the desired sign to be rendered by `@sign`, `@orient`, `@slash`, and `@dot`?

I think this can be closed. The answer to your question is the latter option: the attributes `@modusmaior`, `@modusminor`, `@tempus` and `@prolatio` are meant to _just_ carry the semantics and one always has to include the visual-domain attributes (`@sign`, `@orient`, `@slash`, and `@dot`) to render a mensuration sign. 

This has been clarified in the new version of the guidelines: [5.2.2 Implicit mensuration](https://music-encoding.org/guidelines/dev/content/mensural.html#implicitmensuration). And I doubled-checked that Verovio is doing this too (for `@meiversion="5.0.0-dev"`).

So, yes, we can close this issue. Thank you!
